app/components/money-amount-input/index.tsx

```diff
@@ -50,11 +50,16 @@ export const MoneyAmountInput = ({
   return (
     <FakeCurrencyInput
       value={moneyAmountToMajorUnitOrSats(moneyAmount)}
-      onChangeValue={(amount) =>
-        setAmount &&
-        setAmount(
-          amountInMajorUnitOrSatsToMoneyAmount(Number(amount), moneyAmount.currency),
-        )
+      onChangeValue={
+        setAmount
+          ? (amount) =>
+              setAmount(
+                amountInMajorUnitOrSatsToMoneyAmount(
+                  Number(amount),
+                  moneyAmount.currency,
+                ),
+              )
+          : undefined
```

Comment:
imo, it starts to be complex enough that I would not inline that in the JSX but have this in the code before the return